src/commands/PublishCommand.js

```diff
@@ -338,14 +338,15 @@ export default class PublishCommand extends Command {
     });
 
     this.progressBar.init(this.packagesToPublish.length);
+    this.npmRegistry = this.getOptions().registry;
```

Comment:
Moved to runPreparations() so every command has access to it. I hope it does not count as an overkill :)